### ISSUE TITLE
508-defect-2 [AXE-CORE]: Heading levels SHOULD increase by one

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#12663]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

The alert box heading, "Save time—and save your work in progress—by signing in before starting your application," is currently an h3 but there isn't an h2 between it and the h1. 

If this page is in the CMS, it is a CMS sitewide issue. 

If this page is built in React, there is a level prop that needs to be adjusted to "2", and a utility class will style it to look like an h3.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to navigate the hierarchy of the page content using heading levels to save time and frustration.

## Environment

- Operating System: all
- Browser: all
- Screenreading device: any
- Server destination: staging & production

## Steps to Recreate

1. Enter [URL] in browser
2. Have developer tools open, and the axe browser extension loaded
3. Run an axe audit
4. Verify that [heading code] is called out as an error of "Heading levels should only increase by one"
5. This error is repeated throughout the form process

### Sample 

**Current code**
```html
<h3 class="usa-alert-heading">Save time—and save your work in progress—by signing in before starting your application</h3>
```

**Recommended code**
```html
<h2 class="usa-alert-heading">Save time—and save your work in progress—by signing in before starting your application</h2>
```

## WCAG or Vendor Guidance (optional)

* [axe-core 3.4 - Heading levels should only increase by one](https://dequeuniversity.com/rules/axe/3.4/heading-order)

## Screenshots

![Screen Shot 2020-08-20 at 11 50 56 AM](https://user-images.githubusercontent.com/57469/90794936-706b0b00-e2db-11ea-85c2-27b376d283d4.png)

![Screen Shot 2020-08-20 at 11 47 02 AM](https://user-images.githubusercontent.com/57469/90794532-e622a700-e2da-11ea-8916-850060f7e959.png)

**Comment by Mottie** [comment]
These changes will require fixes in the form system save-in-progress component owned by VSP.

**Comment by jenstrickland** [comment]
@meganhkelley It looks like the save-in-progress component needs to have the level prop adjusted to be `2` with the styling of the h3.

**Comment by cvalarida** [comment]
:wave: Hey, @jenstrickland. Is this a problem for all forms? Only for specific forms? Based on @Mottie's comment, I assume it's affecting everything, in which case it would fall on us for sure.

**Comment by Mottie** [comment]
> I assume it's affecting everything

Yes, here are some example screenshots that I copied over from https://github.com/department-of-veterans-affairs/va.gov-team/issues/12666#issuecomment-690771542

<details><summary>Retrieving your profile information</summary>


![Screen Shot 2020-09-10 at 5 39 42 PM](https://user-images.githubusercontent.com/136959/92818394-47471300-f38d-11ea-9b65-292c4d5e9ce6.png)</details>

<details><summary>Saving your application</summary>


![Screen Shot 2020-09-10 at 5 41 40 PM](https://user-images.githubusercontent.com/136959/92818387-4615e600-f38d-11ea-8d55-5600a7edee10.png)</details>

<details><summary>Retrieving your saved application</summary>


![Screen Shot 2020-09-10 at 5 41 57 PM](https://user-images.githubusercontent.com/136959/92818392-46ae7c80-f38d-11ea-815c-8a47c42bf202.png)</details>

**Comment by jenstrickland** [comment]
@Mottie is it possible to set the heading level prop for this alertbox to 2? Is there something about how the page loads as referenced in the comment with examples from #12666 that impact your access to the prop?

**Comment by Mottie** [comment]
You can set a `level` property on the `<AlertBox level={2} />`, but it's not documented.

**Comment by jenstrickland** [comment]
@Mottie I see it at the bottom of this page: https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/visual-design/components/alertbox/

Were you expecting something else? I think it would be great for the platform to know how to make sure documentation serves devs well.

**Comment by Mottie** [comment]
Oh yeah, the props table is auto-generated, but the code examples don't include the property.

**Comment by andaleliz** [comment]
@cvalarida This does happen on other forms - I noticed this on the [1990 intro page](https://www.va.gov/education/apply-for-education-benefits/application/1990/introduction).

Another thing to note is that authenticated users don't get an `h2` OR an `h3`.

There are some other possible improvements to be made to these form intro pages (e.g. should we really be asking people to sign in/start an application before they've had a chance to review what is required for the form?), that I've captured in #14623 14623.  I don't know exactly who/when this will get tackled but we're capturing these things as we come across them as part of our effort to migrate the form wizards from content pages.

**Comment by taylorkaren** [comment]
Info from @micahchiang: Save in Progress (SiP) is not a priority for the Forms Library Team in Q2, so this will be on the backburner for the time being.
